students/tri_nguyen/lesson04/trigrams.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------- #
+# Student:   Tri Nguyen
+# Instructor: Natasha Aleksandrova
+# Date:  28-Feb-2018
+# ------------------------------------------- #
+
+import random
+
+
+def get_word_list():
+    ''' this function create a list of words from 
+        sherlock_small.txt '''
+
+    word_list = []
+
+    with open('sherlock_small.txt') as f:
+        for line in f:
+            line.rstrip()
+            for char in line:
+                line = line.replace('--', ' ')
+            words = line.split()
+
+            for word in words:
+                word_list.append(word)
+
+    return word_list
+
+
+def create_word_dict():
+    ''' create a dictionary that looks like the
+        trigram table '''
+
+    word_dict = {}
+
+    txt_words = get_word_list()
+
+    for idx in range(len(txt_words) - 2):
+        word_dict.setdefault((txt_words[idx], txt_words[idx + 1]), []).append(txt_words[idx + 2])
```

Comment:
this could be a bit simpler by using `defaultdict` -> `word_dict = defaultdict(list)`

then you have:
```
word_dict((txt_words[idx], txt_words[idx + 1])).append(txt_words[idx + 2])
```